### PR TITLE
Problem: 0.7.x fix-unlucky-tx command does not include concurrency and null indexer

### LINKED ISSUE #542: Problem: 0.7.x fix-unlucky-tx command does not include concurrency and null indexer
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by allthatjazzleo**
fix #542 

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

**Review by yihuang — CHANGES_REQUESTED** [review]
we can also change the `cobra.ExactArgs(1)` -> `cobra.ExactArgs(0)` now.

File: cmd/cronosd/cmd/fix-unlucky-tx.go [hunk 1]
```diff
@@ -122,21 +117,31 @@ func FixUnluckyTxCmd() *cobra.Command {
 							}
 							txResult.Events = append(txResult.Events, evt)
 						}
-						if err := tmDB.stateStore.SaveABCIResponses(blockNumber, blockResults); err != nil {
+
+						if printBlockNumbers {
+							fmt.Println(height, txIndex)
```
[the remark this answers is not in the document]
Comment (allthatjazzleo) [reply cmd/cronosd/cmd/fix-unlucky-tx.go]: do you mean move 
```
						if printBlockNumbers {
							fmt.Println(height)
							return nil
						}
```
before 
```
txIndex++
```
?

File: cmd/cronosd/cmd/fix-unlucky-tx.go [hunk 2]
```diff
@@ -122,21 +117,31 @@ func FixUnluckyTxCmd() *cobra.Command {
 							}
 							txResult.Events = append(txResult.Events, evt)
 						}
-						if err := tmDB.stateStore.SaveABCIResponses(blockNumber, blockResults); err != nil {
+
+						if printBlockNumbers {
+							fmt.Println(height)
```
[the remark this answers is not in the document]
Comment (allthatjazzleo) [reply cmd/cronosd/cmd/fix-unlucky-tx.go:122]: do you mean move 
```
						if printBlockNumbers {
							fmt.Println(height)
							return nil
						}
```
before 
```
txIndex++
```
?

File: cmd/cronosd/cmd/fix-unlucky-tx.go [hunk 3]
```diff
@@ -51,68 +54,67 @@ func FixUnluckyTxCmd() *cobra.Command {
...
+							return fmt.Errorf("block not found: %d", height)
 						}
 
 						tx, err := clientCtx.TxConfig.TxDecoder()(blk.Txs[i])
 						if err != nil {
 							return err
 						}
 
+						if _, ok := tx.GetMsgs()[0].(*evmtypes.MsgEthereumTx); !ok {
+							 // ignore non-eth tx
```
Comment (yihuang) [inline cmd/cronosd/cmd/fix-unlucky-tx.go:110]:
```suggestion
							// ignore non-eth tx
```

**Pull request merged**